atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2465,18 +2491,28 @@ protected ListenableFuture<Map<Long, Long>> getCommitTimestamps(
         if (Iterables.isEmpty(startTimestamps)) {
             return Futures.immediateFuture(ImmutableMap.of());
         }
+
```

Comment:
All of this logic should probably live elsewhere

Reply:
note: Let's start with commitTsLoader that encapsulates the cache as well.